File: algorithms/intersectionTwoArrays/intersectionTwoArrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    var result = [];
    nums1.forEach(function(num) {
        if (result.indexOf(num) >= 0) return;
        if (nums2.indexOf(num) >= 0) result.push(num); 
    });
    return result;
};

File: algorithms/intersectionTwoArrays/intersectionTwoArrays.swift
class Solution {
    func intersection(nums1: [Int], _ nums2: [Int]) -> [Int] {
        var result = [Int]()
        for num in nums1 {
            if result.indexOf(num) != nil {
                continue
            } else if nums2.indexOf(num) != nil {
                result.append(num)
            }
        }
        return result
    }
}